### PR TITLE
feat: add windows images for LTSC 2019

### PR DESCRIPTION
**Pull request by garethjevans**

### CODE REVIEW

File: 11/windows/windowsservercore-2019/openj9/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+# for main web interface:
+EXPOSE ${http_port}
+
+# will be used by attached agents:
+EXPOSE ${agent_port}
+
+ENV COPY_REFERENCE_FILE_LOG $JENKINS_HOME/copy_reference_file.log
+
+# # set variables to create shared class cache
```
Comment (slide) [inline 11/windows/windowsservercore-2019/openj9/Dockerfile:66]: Is there a reason this part is commented out? It creates a class cache that makes subsequent start-ups faster.

Comment (garethjevans) [reply 11/windows/windowsservercore-2019/openj9/Dockerfile:66]: only cause they were originally commented out in the image that I copied - https://github.com/jenkinsci/docker/blob/c151d909879755d18246bca514ec8ce620499633/11/windows/windowsservercore-1809/openj9/Dockerfile#L66-L76

Comment (garethjevans) [reply 11/windows/windowsservercore-2019/openj9/Dockerfile:66]: Would be happy to uncomment them both if you think it would help?

Comment (slide) [reply 11/windows/windowsservercore-2019/openj9/Dockerfile:66]: Yes, that would be good. Thanks!

File: Jenkinsfile [hunk 2]
```diff
@@ -140,7 +140,7 @@ stage('Build') {
 
 void nodeWithTimeout(String label, def body) {
     node(label) {
-        timeout(time: 60, unit: 'MINUTES') {
+        timeout(time: 100, unit: 'MINUTES') {
```
Comment (daniel-beck) [inline Jenkinsfile:143]: No. Just no.

During a security release, I need to wait for the images to be able to ensure ci.jenkins.io is protected before I publish the advisory and push sources (and tests with sample exploits). Parallelize this or or something, but this increasingly taking forever is simply not OK.

**Review by daniel-beck — CHANGES_REQUESTED** [review]

**Comment by daniel-beck** [comment]
I would like to see the data that says there's even a single user for each of the images you're adding here. I suspect there is not.

**Comment by slide** [comment]
> I would like to see the data that says there's even a single user for each of the images you're adding here. I suspect there is not.

In order to run on Google Cloud, Windows docker images have to be at least 1909. We have received requests to run on Google Cloud, so that is where this is coming from. We could probably reduce to just jdk11 and so forth to reduce the amount of time it takes.

**Comment by daniel-beck** [comment]
> In order to run on Google Cloud, Windows docker images have to be at least 1909.

So this could replace the 1809 with 2019, easy. We'd just end up in another #1033 if we keep things around forever.

**Comment by slide** [comment]
not necessarily containers work a little differently on Windows sometimes you need the actual version of Windows that you're running on to be able to use a container

**Comment by slide** [comment]
I would like to propose that we follow something similar to what Adoptium (the artiest formerly known as AdoptOpenJDK). We should have a small set of "official" images that we publish that are the ones that you would need to worry about for security releases and other "unofficial" images that you don't. I'll start a topic about this with the platform SIG.

**Comment by daniel-beck** [comment]
> I would like to propose that we follow something similar to what Adoptium (the artiest formerly known as AdoptOpenJDK). We should have a small set of "official" images that we publish that are the ones that you would need to worry about for security releases and other "unofficial" images that you don't. I'll start a topic about this with the platform SIG.

That seems like an interesting approach. Thank you for taking this on!

Don't we have a weird blueocean image around somewhere that would clearly be unofficial?

**Comment by slide** [comment]
Yes, there is a blue ocean image that is old and not maintained that could possibly be in that category

**Comment by MarkEWaite** [comment]
The blueocean image was used by the "Installing Docker" instructions on
jenkins.io and by three tutorials on jenkins.io until Vlad Silverman's pull
request to use the jenkins/jenkins images about a month ago.

It is not maintained but as far as I know, it is kept current by automated
processes that use the most recent LTS and the most recent Blue Ocean
plugin releases.

I believe the blueocean image should be officially declared "deprecated"
now that we have install instructions and tutorials dependencies fixed to
use jenkins/jenkins instead of the blueocean image.

**Comment by slide** [comment]
FYI, I am adding 20H2 images for Adoptium. 

https://github.com/AdoptOpenJDK/openjdk-docker/pull/472

**Comment by garethjevans** [comment]
@slide 2019 ltsc also appears to be supported on Google Cloud - https://cloud.google.com/compute/docs/images/os-details#windows_server

**Comment by slide** [comment]
I think we can reduce this to just one added image if we just provide a jdk11 hotspot image instead of all four variants. Thoughts?

**Comment by garethjevans** [comment]
This is only "jdk11 hotspot" at the moment, I've removed the other images

**Review by slide — APPROVED** [review]

**Comment by slide** [comment]
@daniel-beck Are you ok adding this image since we actually reduced the number of windows images with my other PR? We would be at two images instead of four if we merge this. I will also look at parallelizing the Windows builds like we have for the Linux side.

**Comment by daniel-beck** [comment]
Is this even being built? I just see a file added, are we iterating over directories for this?

**Comment by slide** [comment]
> Is this even being built? I just see a file added, are we iterating over directories for this?

Yes, it is being built. The make.ps1 file finds all the Windows Dockerfiles and builds them. See https://ci.jenkins.io/blue/organizations/jenkins/Packaging%2Fdocker/detail/PR-1020/98/pipeline/28#step-29-log-169 for instance.

**Comment by slide** [comment]
@daniel-beck  We've reduced the number of Windows builds, are you ok with merging this? This would bring us to 2 when we had 4 before. You have "Changed Requested" in your review and I would like to move forward with this.

File: 11/windows/windowsservercore-2019/hotspot/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# escape=`
+FROM jenkins4eval/openjdk:11-hotspot-windowsservercore-ltsc2019
```
Comment (daniel-beck) [inline 11/windows/windowsservercore-2019/hotspot/Dockerfile:2]: Why are we pretending that `jenkins4eval` is "unofficial" if we end up just repackaging it into `jenkins` images?

Comment (garethjevans) [reply 11/windows/windowsservercore-2019/hotspot/Dockerfile:2]: @daniel-beck these images are not yet published by AdoptOpenJDK, there is an open PR for this but we've had little progress in getting it merged.  When AdoptOpenJDK start supporting these base images we can remove these.

Would you prefer these were published under a different org? I'm not entirely sure what you are recommending here.

For reference - https://github.com/AdoptOpenJDK/openjdk-docker/pull/365

**Pull request merged**